llvm/include/llvm/IR/RepoDefinition.h

```diff
@@ -49,66 +49,112 @@ void set(GlobalObject *GO, DigestType const &D);
 /// GO does not contain the RepoDefinition metadata.
 std::pair<DigestType, bool> get(const GlobalObject *GO);
 
-/// A structure of a global object (GO) information.
+/// A structure of a global object (GO) information, which is used to calculate
+/// the GO's digest.
 struct GOInfo {
   /// The InitialDigest is the hash of an object itself (a function or global
   /// variable) without consideration for any references to other objects.
   DigestType InitialDigest;
-  /// The Contributions of an object (X) are those objects (Y) which
-  /// transitively reference X and where a potential optimisation to either X or
-  /// any of Y may invalidate both.
-  GOVec Contributions;
-  /// The Dependencies of an object are those objects which it transitively
-  /// references but are not Contributions.
+
+  /// When we calculate the initial digest for each global object (GO), we also
+  /// collect the GO's 'Contributions' and 'Dependencies'. The GO's
```

Comment:
This nicely detailed comment talks about collecting both contributions and dependencies, but the former isn't a member of this struct at all but has been moved to a new type called ”detailed info”. Do we really need to split these up?

Reply:
Revert the change back in commit https://github.com/SNSystems/llvm-project-prepo/pull/127/commits/ab8263e2b3d37556b19128219bce6d449965b7ec.